designs/CentralNuGetManagement/CPVM-Project_Package_Version_Override.md

```diff
@@ -0,0 +1,425 @@
+# CPVM Project Package Version Override
+
+Users have indicated that Central Package Version Management (CPVM) needs a mechanism to override the centrally managed versions of packages on a project-by-project basis. Project Package Version Override allows individual projects to import different versions of NuGet packages organized in the same, centralized location.
+
+# Design Goals
+
+1. Even though package versions need to be specified for specific projects, keep the package versions centralized. Avoid situations where solution-level and project-level package version management are mixed in the same repository.
+2. Provide a means for organizing multiple package versions together.
+3. Manage the package versions of transitive dependencies alongside direct dependencies.
+4. Follow the MSBuild approach with minimal changes to existing XML patterns.
+
+# Feature Behavior
+
+- All `<PackageVersion>` should be defined within `Directory.Packages.props`, and `Version` may not be defined on `<PackageReference>` within projects.
+    - Specifying a `Version` on a `<PackageReference>` will throw a build error.
+    - Specifying a `<PackageVersion>` within a project file is an anti-pattern, and it will generate a build time warning. It will not throw an error, and a developer may choose to suppress the warning by specifying `<AllowProjectPackageVersion>true</AllowProjectPackageVersion>` within a `<PropertyGroup>`.
```

Comment:
This requirement is a new requirement not related with the Version Override. 
I am not sure if it can be detected the location of the items; in addition the PV items can imported from custom props files if developers need to go around this rule. I would suggest not have this rule.

Reply:
I think trying to restrict where a item is declared in MSBuild is largely a losing proposition. 

MSBuild's power comes from the fact that everything is processed in a predictable order.